pkg/controllers/restore/controller.go

```diff
@@ -477,6 +476,26 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 	return nil
 }
 
+// customize provides customization of restored resource for edge cases
+func customize(obj *unstructured.Unstructured) {
+	switch obj.GetKind() {
+	case "ServiceAccount":
+		// remove secrets section as referenced secrets will be removed by k8s Token Controller as they are considered orphaned
+		delete(obj.Object, secretsMapKey)
+		logrus.Debugf("Secrets section is ignored in ServiceAccount %s/%s", obj.GetNamespace(), obj.GetName())
+	case "Cluster":
+		// for fleet cluster it needs to be reimported in order to reissue service account token that is no longer valid
+		if obj.GetAPIVersion() == "fleet.cattle.io/v1alpha1" {
+			n, err := rand.Int(rand.Reader, big.NewInt(99999))
+			if err == nil {
+				if err := unstructured.SetNestedField(obj.Object, n.Int64(), "spec", "redeployAgentGeneration"); err != nil {
+					logrus.Warnf("Fleet cluster %s/%s can't be reset to be re-imported", obj.GetNamespace(), obj.GetName())
```

Comment:
I would argue this should be considered a true error and fail import of this object (and consequently the import itself as well). If we somehow restore without this change to CR applied, restored cluster won't be functioning OK.

Reply:
Yeah, I actually don't want this error to fail the whole restore process so that in case error happens, user can patch it manually. This will not likely to cause an error unless the fields are incorrect so I think printing warning might be enough.

Reply:
It is more like conservative so that in case we break the api that this is causing error, user has way to workaround it instead of letting it fail.

Reply:
I see the logic here, but a bit concerned users won't know which message to check for. Though given how unlikely is this error to happen, I think we're OK with the current approach.